### PR TITLE
[OTX] Apply changes in develop to feature/otx branch

### PR DESCRIPTION
**Pull request by goodsong81**
- MPA PR: https://github.com/openvinotoolkit/model_preparation_algorithm/pull/105
- [x] Move changes in external to otx.algorithms
  - [x] Anomaly
  - [x] Classification
  - [x] Detection / instance segmentation
  - [x] Segmentation
  - [x] MPA
- [x] Move changes in ote_sdk to otx.sdk
- [x] Move changes in ote_cli to otx.cli

### CODE REVIEW

**Comment by goodsong81** [comment]
@samet-akcay @ashwinvaidya17 @djdameln 
Changes from develop in external/anomaly look like mainly bug fixes and DRAEM algo.
Could you create a PR to this branch to apply them in otx.algorhtms.anomaly?
We'll take care other tasks.

File: otx/algorithms/classification/tasks/inference.py [hunk 1]
```diff
@@ -346,7 +346,6 @@ def _init_recipe(self):
         # FIXME[Soobee] : if train type is not in cfg, it raises an error in default INCREMENTAL mode.
         # During semi-implementation, this line should be fixed to -> self._recipe_cfg.train_type = train_type
         self._recipe_cfg.train_type = train_type.name
-        patch_data_pipeline(self._recipe_cfg, self.data_pipeline_path)
```
Comment (goodsong81) [inline otx/algorithms/classification/tasks/inference.py]: @supersoob I'm not sure which PR removed this line but could you check if this is OK?

**Comment by goodsong81** [comment]
image

All detection CLI tests failed due to CPU shared memeory size issue.
image
image

I got this solution to increase shared memory size: $ docker run ... --ipc=host ...
Can we try this option? @ashwinvaidya17 @yunchu 
https://discuss.pytorch.org/t/unable-to-write-to-file-torch-18692-1954506624/9990/2

**Comment by ashwinvaidya17** [comment]
I just noticed that there are two containers running. 
image

**Comment by goodsong81** [comment]
> I just noticed that there are two containers running. image

Yunchu ran 2 containers as we have 2 GPUs.
@yunchu Could you re-run the containers with `--ipc=host` option?

**Comment by yunchu** [comment]
> > I just noticed that there are two containers running. image
> 
> Yunchu ran 2 containers as we have 2 GPUs. @yunchu Could you re-run the containers with `--ipc=host` option?

re-started with --ipc=host flag for both runners

**Comment by goodsong81** [comment]
Tests all green (except for coverage report upload)
https://github.com/openvinotoolkit/training_extensions/actions/runs/3790905487/jobs/6446206470
image

**Pull request merged**